airflow/plugins/operators/scrape_ntd_api.py

```diff
@@ -0,0 +1,148 @@
+import csv
+import gzip
+import logging
+import os
+
+# import re
+from typing import Optional
+
+import pendulum
+import requests
+from calitp_data_infra.storage import get_fs  # type: ignore
+from pydantic import BaseModel, Json
+
+from airflow.models import BaseOperator  # type: ignore
+
+# CALITP_BUCKET__NTD_DATA_PRODUCTS = os.environ["CALITP_BUCKET__NTD_DATA_PRODUCTS"]
+
+filename = "./data/ntd_endpoints.csv"
+
+
+def write_to_log(logfilename):
+    """
+    Creates a logger object that outputs to a log file, to the filename specified,
+    and also streams to console.
+    """
+    logger = logging.getLogger(__name__)
+    logger.setLevel(logging.INFO)
+    formatter = logging.Formatter(
+        "%(asctime)s:%(levelname)s: %(message)s", datefmt="%y-%m-%d %H:%M:%S"
+    )
+    file_handler = logging.FileHandler(logfilename)
+    file_handler.setFormatter(formatter)
+    stream_handler = logging.StreamHandler()
+    stream_handler.setFormatter(formatter)
+
+    if not logger.hasHandlers():
+        logger.addHandler(file_handler)
+        logger.addHandler(stream_handler)
+
+    return logger
+
+
+class NtdDataProductAPIExtract(BaseModel):
+    endpoint_id: str
+    product: str
+    year: str
+    data: Json
+    logger: Optional[logging.Logger]
+    extract_time: Optional[pendulum.DateTime]
+
+    # bucket = CALITP_BUCKET__NTD_DATA_PRODUCTS
+    logger = write_to_log("load_ntd_apidata_output.log")
+    extract_time = pendulum.now()
+
+    # pydantic doesn't know dataframe type
+    # see https://stackoverflow.com/a/69200069
+    class Config:
+        arbitrary_types_allowed = True
+
+    def fetch_from_ntd_api(self):
+        """Restore comments"""
+
+        self.logger.info(f"Downloading NTD data for {self.year} / {self.product}.")
+
+        url = (
+            "https://data.transportation.gov/resource/"
+            + self.endpoint_id
+            + ".json"
+            + "?$limit=5000000"
+        )
+
+        response = requests.get(url).content
+
+        if response is None or len(response) == 0:
+            self.logger.info(
+                f"There is no data to download for {self.year} / {self.product}. Ending pipeline."
+            )
+            pass
+        else:
+            self.data = response
+            self.logger.info(
+                f"Downloaded {self.product} data for {self.year} with {len(self.data)} rows!"
+            )
```

Comment:
I don't know anything offhand about the failure modes of the NTD API, but it might be useful to check response status and log non-200 responses or something.